src/server/rpc/procedures/baidu-maps/baidu-maps.js

```diff
@@ -0,0 +1,357 @@
+                return cacheCallback(null, result);
+            });
+        }, (err, imageBuffer) => {
+            // Send the response to the user
+            logger.trace('Sending the response!');
+            // Set the headers
+            response.set('cache-control', 'private, no-store, max-age=0');
+            response.set('content-type', 'image/png');
+            response.set('content-length', imageBuffer.length);
+            response.set('connection', 'close');
+
+            response.status(200).send(imageBuffer);
+            logger.trace('Sent the response!');
+        });
+
+    });
+};
+
+/**
+ * Get a map image of the given region.
+ * @param {Latitude} latitude Latitude of center point
+ * @param {Longitude} longitude Longitude of center point
+ * @param {BoundedNumber<0,1024>} width Image width
+ * @param {BoundedNumber<0,1024>} height Image height
+ * @param {BoundedNumber<1,19>} zoom Zoom level of map image
+ * @returns {Image} Map image
+ */
+BaiduMaps.prototype.getMap = function(latitude, longitude, width, height, zoom){
+
+    this._getMap(latitude, longitude, width, height, zoom);
+
+    return null;
+};
+
+/**
+ * Convert longitude to the x value on the map image.
+ * @param {Longitude} longitude Longitude coordinate
+ * @returns {Number} Map x coordinate of the given longitude
+ */
+BaiduMaps.prototype.getXFromLongitude = function(longitude) {
+    return this._getUserMap(this.caller.clientId).then(mapInfo => {
+        let pixels = this._pixelsAt(0,longitude, mapInfo);
+        return pixels.x;
+    });
+};
+
+/**
+ * Convert latitude to the y value on the map image.
+ * @param {Latitude} latitude Latitude coordinate
+ * @returns {Number} Map y coordinate of the given latitude
+ */
+BaiduMaps.prototype.getYFromLatitude = function(latitude) {
+    return this._getUserMap(this.caller.clientId).then(mapInfo => {
+        let pixels = this._pixelsAt(latitude,0, mapInfo);
+        return pixels.y;
+    });
+};
+
+/**
+ * Convert x value of map image to longitude.
+ * @param {Number} x x value of map image
+ * @returns {Longitude} Longitude of the x value from the image
+ */
+BaiduMaps.prototype.getLongitudeFromX = function(x){
+    return this._getUserMap(this.caller.clientId).then(mapInfo => {
+        let coords = this._coordsAt(x,0, mapInfo);
+        return coords.lon;
+    });
+};
+
+/**
+ * Convert y value of map image to latitude.
+ * @param {Number} y y value of map image
+ * @returns {Latitude} Latitude of the y value from the image
+ */
+BaiduMaps.prototype.getLatitudeFromY = function(y){
+    return this._getUserMap(this.caller.clientId).then(mapInfo => {
+        let coords = this._coordsAt(0,y, mapInfo);
+        return coords.lat;
+    });
+};
+
+/**
+ * Convert x value of map image to longitude.
+ * @param {Number} x x value of map image
+ * @returns {Longitude} Longitude of the x value from the image
+ *
+ * @deprecated
+ */
+BaiduMaps.prototype.getLongitude = function(x){
```

Comment:
We don't need to support deprecated methods for BaiduMaps

Reply:
They're gone now.